src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -192,6 +192,28 @@ public void visit(NodeTraversal t, Node n, Node parent) {
         visitRequireCall(t, n, parent);
       }
 
+      // Finds calls to require.ensure which is a method to allow async loading of
+      // CommonJS modules:
+      //
+      // require.ensure(['/path/to/module1', '/path/to/module2'], function(require) {
+      //    var module1 = require('/path/to/module1');
+      //    var module2 = require('/path/to/module2');
+      // });
+      //
+      // will be rewritten as an IIFE
+      //
+      // (function() {
+      //   var module1 = require('/path/to/module1');
+      //   var module2 = require('/path/to/module2');
+      // })()
+      //
+      // See http://wiki.commonjs.org/wiki/Modules/Async/A
+      //     http://www.injectjs.com/docs/0.7.x/cjs/require.ensure.html
+      if (n.isCall() && n.getChildCount() == 3 && n.getFirstChild().isGetProp()
+          && n.getFirstChild().matchesQualifiedName("require.ensure")
+          && n.getSecondChild().isArrayLit() && n.getChildAtIndex(2).isFunction()) {
```

Comment:
In the future, I plan on only removing this call if all of the required modules are provided as source. Once we have the concept of external modules, then this call would need to be left at least partially intact if at least one of the module references in the array literal was external. We won't be able to rewrite calls where the first argument is anything but an array literal.

Also, the second argument must be a function with exactly one parameter named `require`. Aliasing `require` is not something I wanted to tackle right now.

Most of these checks could be deferred to the implementation of `visitRequireEnsureCall` though.